internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -282,6 +345,51 @@ func handleParseError(typ string, parseErr error) *metrics.WebhookEventState {
 	return state
 }
 
+func (_ *Server) parseGithubAppEventForProcessing(
+	rawWHPayload []byte,
+	msg *message.Message,
+) error {
+	var event github.InstallationEvent
+
+	if msg.Metadata.Get(events.GithubWebhookEventTypeKey) != "installation" {
+		return newErrNotHandled("github app event %s not handled", msg.Metadata.Get(events.GithubWebhookEventTypeKey))
+	}
+
+	err := json.Unmarshal(rawWHPayload, &event)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling payload: %w", err)
+	}
+
+	action := event.GetAction()
+	if action == "" {
+		return fmt.Errorf("action is empty")
+	}
+
+	if action != "deleted" {
+		return newErrNotHandled("event %s with action %s not handled",
+			msg.Metadata.Get(events.GithubWebhookEventTypeKey), action)
+	}
+
+	installationID := event.GetInstallation().GetID()
+	if installationID == 0 {
+		return fmt.Errorf("installation ID is 0")
+	}
+
+	payloadBytes, err := json.Marshal(providers.GitHubAppInstallationDeletedPayload{
+		InstallationID: installationID,
+	})
+	if err != nil {
+		return fmt.Errorf("error marshalling payload: %w", err)
+	}
+
+	providers.ProviderInstanceRemovedMessage(
+		msg,
+		db.ProviderClassGithubApp,
+		payloadBytes)
```

Comment:
This feels weird... the function seems to be acting on a side-effect, but the method name is "Provider...Message", not "Send..." or "Publish..." or the like.